OpenRA.Mods.Common/Activities/DonateCash.cs

```diff
@@ -18,24 +18,34 @@ namespace OpenRA.Mods.Common.Activities
 {
 	class DonateCash : Enter
 	{
-		readonly int payload;
-		readonly int playerExperience;
+		readonly DeliversCash deliversCash;
 
-		public DonateCash(Actor self, Target target, int payload, int playerExperience)
+		public DonateCash(Actor self, Target target, DeliversCash deliversCash)
 			: base(self, target, Color.Yellow)
 		{
-			this.payload = payload;
-			this.playerExperience = playerExperience;
+			this.deliversCash = deliversCash;
+		}
+
+		protected override bool TryStartEnter(Actor self, Actor targetActor)
+		{
+			var enterAcceptsCash = targetActor.TraitsImplementing<AcceptsDeliveredCash>().FirstEnabledTraitOrDefault();
+			if (enterAcceptsCash == null || enterAcceptsCash.IsTraitDisabled || deliversCash.IsTraitDisabled)
```

Comment:
`enterAcceptsCash.IsTraitDisabled` should never be true, since we use `FirstEnabledTraitOrDefault`, right?